### ISSUE TITLE
Unable to use ClassicToolchain in explicit way

### ISSUE THREAD

**Issue by ig-sinicyn** [dotnet/BenchmarkDotNet#327]
BDN 10.1, code:
```cs
	public class Program
	{
		public static void Main(string[] args)
		{
			var c = DefaultConfig.Instance.With(
				Job.Default.With(ClassicToolchain.Instance));

			BenchmarkRunner.Run<Program>(c);

			Console.ReadKey();
		}

		[Benchmark]
		public void Test() => Thread.Sleep(1);

		[Benchmark]
		public void Test2() => Thread.Sleep(0);
	}
```

output:
```
// *** Build ***
BuildScript: C:\Users\igors\documents\visual studio 2017\Projects\ConsoleApp2\ConsoleApp2\bin\Release\BDN.Generated.bat
CS0234: The type or namespace name 'RoslynToolchain' does not exist in the namespace 'BenchmarkDotNet.Toolchains' (are you missing an assembly reference?)
// Result = Failure
```
[log.zip](https://github.com/dotnet/BenchmarkDotNet/files/655265/log.zip)

P.S. Possible StackOverflow at [ToolchainExtensions](https://github.com/dotnet/BenchmarkDotNet/blob/0231157fb11d054fcb99474fbba999be22aa7fc2/src/BenchmarkDotNet/Toolchains/ToolchainExtensions.cs#L16)

**UPD**:
Reference to the `BenchmarkDotNet.Toolchains.Roslyn` missed. #234 strikes back :)

**Comment by ig-sinicyn** [comment]
**UPD:** `Net46Toolchain` fails too but the error is different:
```cs
		public static void Main(string[] args)
		{
			var bindingFlags = BindingFlags.Instance | BindingFlags.Public | BindingFlags.NonPublic;
			var toolchain =(Net46Toolchain)typeof(Net46Toolchain)
				.GetConstructors(bindingFlags)[0]
				.Invoke(new object[0]);
			Console.WriteLine(ClassicToolchain.Instance.GetType());
			var c = DefaultConfig.Instance.With(
				Job.Default.With(toolchain));

			BenchmarkRunner.Run<Program>(c);

			Console.ReadKey();
		}
```
=>
```
// *** Build ***
// Result = Failure
// Exception: dotnet restore has failed
```
(Fresh windows 10 + VS 2017 RC installed).
[log.zip](https://github.com/dotnet/BenchmarkDotNet/files/655372/log.zip)

**Comment by ig-sinicyn** [comment]
Still reproduces with Net46Toolchain and CoreToolchain. VS 2017.
```
exception:  dotnet restore has failed
```
[log.zip](https://github.com/dotnet/BenchmarkDotNet/files/747873/log.zip)

**Comment by ig-sinicyn** [comment]
UPD: more detail : .net 46 toolchain emits `project.json` and it looks like latest dotnet cli (1.1.0) [does not support it anymore](https://github.com/dotnet/core/issues/440). 

archive with generated project:
[Program_Test_Job-ZKILXX.zip](https://github.com/dotnet/BenchmarkDotNet/files/747901/Program_Test_Job-ZKILXX.zip)

**Comment by adamsitnik** [comment]
The real bug was caused by recent characteristic changes, which required parameterless ctors for toolchains. [Fix](https://github.com/dotnet/BenchmarkDotNet/commit/5c8c75d0ab8cc15b2b908395416fe04a9613fc5f)

The thing you reported now is lack of support for latest dotnet cli, which is tracked as part of #300 . I am closing this one, will be fixed as part of #300
